_posts/2022-07-20-polymorphic-table-functions.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: post
+title: "Diving into polymorphic table functions with Trino"
+author: "Kasia Findeisen, Brian Olsen, and Cole Bowden"
+excerpt_separator: <!--more-->
+---
+
+In the Trino community, we know that being the coolest query engine is a tough
+job. We boldly face the intricacies of the SQL standard to bring you the newest
+and most powerful features. Today, we proudly announce that as of release 381,
+Trino is on its way to full support for polymorphic table functions (PTFs).
+
+In this blog post, we are explaining the concept of table functions and 
+exploring how they can be leveraged. We also look at what we have already 
+implemented, and take a sneak peek into the future.
+
+<!--more-->
+
+### Definition time
+
+There are several kinds of functions you can call in a SQL query: scalar
+functions, aggregate functions, and window functions. They might process the
+input row by row (scalar) or all at once (aggregate). One thing they have in
+common is that they return scalar values. Table functions are different. They
+return tables. In a query, they can appear in any place where a table reference
+shows up such as a `FROM` clause:
+
+```sql
+SELECT
+  *
+FROM
+  TABLE(my_table_function('foo'));
+```
+
+You can also use table functions in joins:
+
+```sql
+SELECT
+  *
+FROM
+  TABLE(my_table_function('bar'))
+JOIN
+  TABLE(another_table_funciton(1, 2, 3))
+ON true;
+```
+
+Polymorphic table functions (PTFs) are a subset of table functions where the
+schema of the returned table is determined dynamically. The returned table
+schema can depend on the arguments you pass to the function.
+
+### OK, but why are we so excited?
+
+We are excited because this feature is a real game changer! Polymorphic table
+functions make SQL extensible and act as a bridge between the Trino engine and
```

Comment:
“act as a bridge…” is a mischaracterization. That’s one possible application, but there are functions one could write that have nothing to do with external systems or resources, much like any scalar, aggregation or window function